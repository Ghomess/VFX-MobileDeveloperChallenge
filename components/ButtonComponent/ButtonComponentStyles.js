import {StyleSheet} from 'react-native';

export const buttonStyles = StyleSheet.create({
  button: {
    backgroundColor: '#1A237E',
    padding: 20,
    borderRadius: 10,
  },
  title: {
    color: '#ffff',
    fontSize: 20,
  },
});
