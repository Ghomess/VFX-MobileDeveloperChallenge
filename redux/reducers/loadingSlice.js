import {createSlice} from '@reduxjs/toolkit';

const initialState = {
  loadingChart: false,
  loadingSearchResults: false,
};

const loadingSlice = createSlice({
  name: 'loading',
  initialState,
  reducers: {
    changeLoadingChart: (state, action) => {
      console.log('CHANGE loadingChart: ', action.payload);
      state.loadingChart = action.payload;
    },
    changeLoadingSearchResults: (state, action) => {
      console.log('CHANGE loadingSearchResults: ', action.payload);
      state.loadingSearchResults = action.payload;
    },
  },
});

export const {changeLoadingChart, changeLoadingSearchResults} =
  loadingSlice.actions;

export default loadingSlice.reducer;
