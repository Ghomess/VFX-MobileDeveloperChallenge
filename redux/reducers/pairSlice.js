import {createSlice} from '@reduxjs/toolkit';

const initialState = {
  pair: null,
  pairData: [],
  pairDateSelected: '',
  pairDateType: 'Daily',
};

const pairSlice = createSlice({
  name: 'pair',
  initialState,
  reducers: {
    addPair: (state, action) => {
      console.log('ADD PAIR: ', action.payload);
      state.pair = action.payload;
    },
    addPairData: (state, action) => {
      console.log('ADD PAIR DATA: ', action.payload);
      state.pairData = action.payload;
    },
    addPairDateSelected: (state, action) => {
      console.log('ADD PAIR DATE: ', action.payload);
      state.pairDateSelected = action.payload;
    },
    addPairDateType: (state, action) => {
      console.log('ADD PAIR DATE: ', action.payload);
      state.pairDateType = action.payload;
    },
  },
});

export const {addPair, addPairData, addPairDateSelected, addPairDateType} =
  pairSlice.actions;

export default pairSlice.reducer;
