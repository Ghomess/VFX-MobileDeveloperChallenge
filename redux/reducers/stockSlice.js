import {createSlice} from '@reduxjs/toolkit';

const initialState = {
  ticker: null,
  stockData: {},
  tickerSearchResults: [],
  stockMonthSelected: null,
  stockDateType: 'Monthly',
  stockPrice: null,
};

const stockSlice = createSlice({
  name: 'ticker',
  initialState,
  reducers: {
    addticker: (state, action) => {
      console.log('ADD ticker: ', action.payload);
      state.ticker = action.payload;
    },
    addstockData: (state, action) => {
      console.log('ADD stock DATA: ', action.payload);
      state.stockData = action.payload;
    },
    addstockMonthSelected: (state, action) => {
      console.log('ADD stock MONTH: ', action.payload);
      state.stockMonthSelected = action.payload;
    },
    addtickerSearchResults: (state, action) => {
      console.log('ADD ticker SEARCH RESULTS: ', action.payload);
      state.tickerSearchResults = action.payload;
    },
    addstockPrice: (state, action) => {
      console.log('ADD stock PRICE: ', action.payload);
      state.stockPrice = action.payload;
    },
  },
});

export const {
  addticker,
  addstockData,
  addstockMonthSelected,
  addtickerSearchResults,
  addstockPrice,
} = stockSlice.actions;

export default stockSlice.reducer;
