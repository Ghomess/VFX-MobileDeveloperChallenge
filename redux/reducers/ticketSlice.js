import {createSlice} from '@reduxjs/toolkit';

const initialState = {
  ticket: null,
  ticketData: {},
  ticketSearchResults: [],
  ticketMonthSelected: null,
};

const ticketSlice = createSlice({
  name: 'ticket',
  initialState,
  reducers: {
    addTicket: (state, action) => {
      console.log('ADD TICKET: ', action.payload);
      state.ticket = action.payload;
    },
    addTicketData: (state, action) => {
      console.log('ADD TICKET DATA: ', action.payload);
      state.ticketData = action.payload;
    },
    addTicketMonth: (state, action) => {
      console.log('ADD TICKET MONTH: ', action.payload);
      state.ticketMonthSelected = action.payload;
    },
    addTicketSearchResults: (state, action) => {
      console.log('ADD TICKET SEARCH RESULTS: ', action.payload);
      state.ticketSearchResults = action.payload;
    },
  },
});

export const {
  addTicket,
  addTicketData,
  addTicketMonth,
  addTicketSearchResults,
} = ticketSlice.actions;

export default ticketSlice.reducer;
