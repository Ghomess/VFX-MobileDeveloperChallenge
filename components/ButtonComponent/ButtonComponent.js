import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {buttonStyles} from './ButtonComponentStyles';
import {useDispatch, useSelector} from 'react-redux';
import {addPairDateType} from '../../redux/reducers/pairSlice';

import {useNavigation} from '@react-navigation/native';
import {changeLoadingChart} from '../../redux/reducers/loadingSlice';

const ButtonComponent = ({title}) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const navigationState = navigation.getState();
  const currentScreen = navigationState.routes[navigationState.index].name;
  const pairDateType = useSelector(state => state.pair.pairDateType);
  const stockDateType = useSelector(state => state.ticker.stockDateType);
  const checkDataType = () => {
    if (currentScreen === 'Stocks') {
      return stockDateType;
    } else if (currentScreen === 'Currency') {
      return pairDateType;
    }
  };

  return (
    <TouchableOpacity
      onPress={() => {
        dispatch(changeLoadingChart(true));
        dispatch(addPairDateType(title));
      }}
      style={[
        buttonStyles.button,
        checkDataType() === title && buttonStyles.buttonSelected,
      ]}>
      <Text
        style={[
          buttonStyles.title,
          checkDataType() === title && buttonStyles.titleSelected,
        ]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

export default ButtonComponent;
