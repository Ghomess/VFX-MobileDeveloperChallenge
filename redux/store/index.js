import {configureStore} from '@reduxjs/toolkit';
import stockSlice from '../reducers/stockSlice';
import pairSlice from '../reducers/pairSlice';
import loadingSlice from '../reducers/loadingSlice';

const store = configureStore({
  reducer: {
    ticker: stockSlice,
    pair: pairSlice,
    loading: loadingSlice,
  },
});

export default store;
