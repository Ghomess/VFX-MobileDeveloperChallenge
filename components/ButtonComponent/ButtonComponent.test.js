import React from 'react';

import ButtonComponent from './ButtonComponent';
import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import store from '../../redux/store';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {changeLoadingChart} from '../../redux/reducers/loadingSlice';
import {addPairDateType} from '../../redux/reducers/pairSlice';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    getState: () => ({
      routes: [{name: 'Currency'}],
      index: 0,
    }),
  }),
}));

describe('ButtonComponent', () => {
  test('renders correctly', async () => {
    await waitFor(() => {
      const tree = render(
        <Provider store={store}>
          <NavigationContainer>
            <ButtonComponent title={'Daily'} />
          </NavigationContainer>
        </Provider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  test('dispatches addPairDateType action of the title on press', async () => {
    await waitFor(() => {
      const dispatchMock = jest.spyOn(store, 'dispatch');
      const title = 'Daily';
      const {getByTestId} = render(
        <Provider store={store}>
          <NavigationContainer>
            <ButtonComponent title={title} />
          </NavigationContainer>
        </Provider>,
      );
      const touchable = getByTestId('ButtonComponent.ToucableOpacity');
      fireEvent.press(touchable);
      expect(dispatchMock).toHaveBeenCalledWith(addPairDateType(title));
    });
  });

  test('dispatches changeLoadingChart action on press', async () => {
    await waitFor(() => {
      const dispatchMock = jest.spyOn(store, 'dispatch');
      const {getByTestId} = render(
        <Provider store={store}>
          <NavigationContainer>
            <ButtonComponent title={'Daily'} />
          </NavigationContainer>
        </Provider>,
      );
      const touchable = getByTestId('ButtonComponent.ToucableOpacity');
      fireEvent.press(touchable);
      expect(dispatchMock).toHaveBeenCalledWith(changeLoadingChart(true));
    });
  });
});
