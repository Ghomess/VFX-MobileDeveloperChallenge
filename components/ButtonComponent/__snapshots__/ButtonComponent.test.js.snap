// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#1A237E",
      "borderRadius": 10,
      "borderWidth": 2,
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffff",
          "fontSize": 20,
        },
        {
          "color": "#1A237E",
        },
      ]
    }
  >
    Daily
  </Text>
</View>
`;
